docs/reference/configuration/single-sign-on/application-load-balancer.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Application Load Balancer
+---
+
+UDS Core supports external TLS termination and custom client certificate headers for environments using external load balancers (e.g., AWS Application Load Balancer - ALB). This is achieved by overriding default Keycloak and Istio settings in your bundle configuration.
+
+The configuration requires disabling HTTPS redirects in Istio gateways and setting the number of trusted proxies to one, with the ALB acting as the trusted proxy. Changing this setting in runtime requires restarting the Istio Tenant Gateway Pods (using `uds zarf tools kubectl rollout restart deployment tenant-ingressgateway -n istio-tenant-gateway`). Keycloak configuration is then adjusted by deploying an additional `EnvoyFilter` that converts the Amazon ALB client certificate into a format that Keycloak can understand. The example below outlines the configuration needed to set up the [Amazon ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/mutual-authentication.html) for UDS Core:
+
+```yaml
+packages:
+  - name: core
+    repository: oci://ghcr.io/defenseunicorns/packages/uds/core
+    ref: x.x.x
+    overrides:
+      istio-tenant-gateway:
+        uds-istio-config:
+          values:
+            - path: tls.servers.keycloak.enableHttpsRedirect
+              value: false
+            - path: tls.servers.tenant.enableHttpsRedirect
+              value: false
```

Comment:
Should this also be documented for the admin gateway?

Reply:
Right now, there's a paragraph about both Gateways in `docs/reference/configuration/service-mesh/ingress.md`:

> UDS Core Gateways can be used with Application Load Balancers (ALBs) connected to one of both Istio Gateways. This allows you to leverage the ALB's capabilities for routing, SSL termination, and other features while still using Istio for service mesh functionality

Would you rather handle this separately in the docs somehow? I also think the ALB should cover only Tenant Gateway. Exposing the Admin Gateway with this approach is possible but questionable from the security perspective.

Reply:
I updated the example and added a commented code for the Admin Gateway. I wouldn't like to make it look like a regular (default or expected) configuration.